lib/geordi/dump_loader.rb

```diff
@@ -6,24 +6,43 @@
 module Geordi
   class DumpLoader
 
-    def initialize(file)
+    def initialize(file, database)
       @dump_file = file
+      @database = database
     end
 
     def development_database_config
+      return @config if @config
+
       require 'yaml'
 
       evaluated_config_file = ERB.new(File.read('config/database.yml')).result
 
       # Allow aliases and a special set of classes like symbols and time objects
       permitted_classes = [Symbol, Time]
-      @config ||= if Gem::Version.new(Psych::VERSION) >= Gem::Version.new('3.1.0')
+      database_config = if Gem::Version.new(Psych::VERSION) >= Gem::Version.new('3.1.0')
         YAML.safe_load(evaluated_config_file, permitted_classes: permitted_classes, aliases: true)
       else
         YAML.safe_load(evaluated_config_file, permitted_classes, [], true)
       end
 
-      @config['development']
+      development_config = database_config['development']
+
+      if development_config.values[0].is_a? Hash # Multi-db setup
+        if @database
+          @config = development_config[@database] || Interaction.fail(%(Unknown development database "#{@database}".))
+        else
+          Interaction.fail 'Multiple databases detected. Select the database to load into with `-L database`'
+        end
+      else # Single-db setup
+        if @database
+          Interaction.fail %(Could not select "#{@database}" database in a single-db environment.)
```

Comment:
```suggestion
          Interaction.fail %(Could not select "#{@database}" database in a single-db setup.)
```
(The environment is always "development" locally.)